backend/src/controllers/projects/contacts.js

```diff
@@ -11,19 +12,35 @@ const controller = useController(model, what, "projects");
  * @returns {object}
  */
 controller.updateContacts = async (request, reply) => {
-  let contacts = [];
-  for (const [key, value] of Object.entries(request.body)) {
-    value.map((row) => {
-      contacts.push({
-        contact_role: Number(key),
-        project_id: Number(request.params.id),
-        contact_id: row.value,
-      });
-    });
-  }
-
   try {
-    const result = await model.updateOne(contacts, Number(request.params.id));
+    let contactsFormatted = [];
+    for (const [key, contactsRaw] of Object.entries(request.body)) {
+      if (contactsRaw?.value) {
+        contactsFormatted.push({
+          contact_role: 6,
+          project_id: Number(request.params.id),
+          contact_id: contactsRaw.value,
+        });
+      } else {
+        contactsRaw?.map((row) => {
+          contactsFormatted.push({
+            contact_role: Number(key),
+            project_id: Number(request.params.id),
+            contact_id: row.value,
+          });
+        });
+      }
+    }
+    const projectManager = contactsFormatted.find((contact) => 6 === contact.contact_role);
+    if (projectManager) {
+      await projectsModel.updateOne(
+        {
+          project_manager: projectManager.contact_id,
+        },
+        Number(request.params.id)
```

Comment:
`Number(request.params.id)` is repeated 4 times in this function. It could be done once and stored in a constant.

Reply:
fixed, thanks for the input!